tests/components/light/test_init.py

```diff
@@ -2,495 +2,623 @@
+    call = turn_off_calls[-1]
 
-    def test_services(self):
-        """Test the provided services."""
-        platform = getattr(self.hass.components, "test.light")
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TURN_OFF
+    assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+    assert call.data[light.ATTR_TRANSITION] == "transition_val"
 
-        platform.init()
-        assert setup_component(
-            self.hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
-        )
-        self.hass.block_till_done()
+    # Test toggle
+    toggle_calls = async_mock_service(hass, light.DOMAIN, SERVICE_TOGGLE)
 
-        ent1, ent2, ent3 = platform.ENTITIES
+    await hass.services.async_call(
+        light.DOMAIN,
+        SERVICE_TOGGLE,
+        {ATTR_ENTITY_ID: "entity_id_val", light.ATTR_TRANSITION: "transition_val"},
+        blocking=True,
+    )
 
-        # Test init
-        assert light.is_on(self.hass, ent1.entity_id)
-        assert not light.is_on(self.hass, ent2.entity_id)
-        assert not light.is_on(self.hass, ent3.entity_id)
+    assert len(toggle_calls) == 1
+    call = toggle_calls[-1]
 
-        # Test basic turn_on, turn_off, toggle services
-        common.turn_off(self.hass, entity_id=ent1.entity_id)
-        common.turn_on(self.hass, entity_id=ent2.entity_id)
+    assert call.domain == light.DOMAIN
+    assert call.service == SERVICE_TOGGLE
+    assert call.data[ATTR_ENTITY_ID] == "entity_id_val"
+    assert call.data[light.ATTR_TRANSITION] == "transition_val"
 
-        self.hass.block_till_done()
 
-        assert not light.is_on(self.hass, ent1.entity_id)
-        assert light.is_on(self.hass, ent2.entity_id)
+async def test_services(hass):
+    """Test the provided services."""
+    platform = getattr(hass.components, "test.light")
 
-        # turn on all lights
-        common.turn_on(self.hass)
+    platform.init()
+    assert await async_setup_component(
+        hass, light.DOMAIN, {light.DOMAIN: {CONF_PLATFORM: "test"}}
+    )
+    await hass.async_block_till_done()
 
-        self.hass.block_till_done()
+    ent1, ent2, ent3 = platform.ENTITIES
 
-        assert light.is_on(self.hass, ent1.entity_id)
-        assert light.is_on(self.hass, ent2.entity_id)
-        assert light.is_on(self.hass, ent3.entity_id)
+    # Test init
+    assert light.is_on(hass, ent1.entity_id)
+    assert not light.is_on(hass, ent2.entity_id)
+    assert not light.is_on(hass, ent3.entity_id)
 
-        # turn off all lights
-        common.turn_off(self.hass)
+    # Test basic turn_on, turn_off, toggle services
+    await hass.services.async_call(
+        light.DOMAIN, SERVICE_TURN_OFF, {ATTR_ENTITY_ID: ent1.entity_id}, blocking=True
+    )
+    await hass.services.async_call(
+        light.DOMAIN, SERVICE_TURN_ON, {ATTR_ENTITY_ID: ent2.entity_id}, blocking=True
+    )
 
-        self.hass.block_till_done()
+    assert not light.is_on(hass, ent1.entity_id)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6239266)